src/Tests/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/RazorSourceGeneratorTests.cs

```diff
@@ -944,6 +944,66 @@ public async Task SourceGenerator_DoesNotAddAnyGeneratedSources_WhenSourceGenera
             Assert.Empty(result.GeneratedSources);
         }
 
+        [Fact]
+        public async Task SourceGenerator_CorrectlyGeneratesSourcesOnceSuppressRazorSourceGeneratorIsUnset()
+        {
+            // Regression test for https://github.com/dotnet/aspnetcore/issues/36227
+            // Arrange
+            var project = CreateTestProject(new()
+            {
+                ["Pages/Index.razor"] = "<h1>Hello world</h1>",
+                ["Pages/Counter.razor"] =
+@"
+@using Microsoft.AspNetCore.Components
+@using Microsoft.AspNetCore.Components.Web
+<h1>Counter</h1>
+<button @onclick=""IncrementCount"">Click me</button>",
```

Comment:
This code looks invalid, since `IncrementCount` is not declared.

Is the test OK regardless since we're only running the RSG and not the C# compiler on the result? If so that makes sense.

Reply:
Yup. It's only running the generator driver in a loop (and not the entire compilation). I could use a lambda so it doesn't look like a error:

```suggestion
<button @onclick=""@(() => {})"">Click me</button>",
```